mantidproject/mantid
# Speed Up Color Fill Plot for Point Data

A fix to avoid re-calculating the `BinEdges` for every call to `MatrixWorkspace::getSignalAtCoord`. This is a fix for the issue, but longer term it would be preferable to have a nicer method in the appropriate `Axis` class to do this in a robust way.

More tests were added to check the intermediate cases.

**To test:**

Plot some point data workspaces using the color fill plot option. Check the performance is similar to that for plotting histogram workspaces. For example try the workspaces from the script below.
```
ws = CreateSampleWorkspace(
    Function='Flat background',
    NumBanks=10,
    BankPixelWidth=100)

ws2 = ConvertToPointData(ws)
```

Fixes #20454. 

*Does not need to be in the release notes.* - not a behavioural change or user reported bug.
---

#### Reviewer ####

Please comment on the following ([full description](http://www.mantidproject.org/Individual_Ticket_Testing)):

##### Code Review #####


##### Functional Tests #####


Does everything look good? Mark the review as **Approve**. A member of `@mantidproject/gatekeepers` will take care of it.
